### PR TITLE
first MVP with baseline model on Jun 13th

### PR DESCRIPTION
**Pull request by ruchen01**
My first ever attempt for a pull request!

### CODE REVIEW

File: src/extract_entity.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    #nlp = spacy.load("en_core_web_sm")
+    nlp = en_core_web_sm.load()
+    doc = nlp(data)
+    entity = [(X.text, X.label_) for X in doc.ents]
+    return entity
+
+def export_entity_csv(data_path, csv_dir):
+    data = find_entity(data_path)
+
```
Comment (astromz) [inline src/extract_entity.py]: remove extra line here. A typical convention is two empty lines between classes and functions, but only one empty line for functions within a class.

File: src/extract_entity.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import spacy
+from spacy import displacy
+from collections import Counter
+import en_core_web_sm
+import csv
+
+#data_path = '../data/preprocessed/13_01_000002/1.txt'
+
+def find_entity(data_path):
+
```
Comment (astromz) [inline src/extract_entity.py]: Would be better to have at least a one-liner docstr, e.g., """Find entity and returns a tuple of (text, label)"""

File: src/extract_topic.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+class LemmaCountVectorizer(CountVectorizer):
+    def build_analyzer(self):
+        lemm = WordNetLemmatizer()
+        analyzer = super(LemmaCountVectorizer, self).build_analyzer()
+        return lambda doc: (lemm.lemmatize(w) for w in analyzer(doc))
+
+
+def process_text2vec(text):
+
```
Comment (astromz) [inline src/extract_topic.py]: remove extra line

File: src/extract_topic.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+def generate_text_li(text_dir):
+    text_li = glob.glob(text_dir+'/*.txt')
+    text_data = []
+    for file_name in text_li:
+        with open(file_name) as f:
+            #data = f.readlines()
+            data = " ".join(line.strip() for line in f)
+            text_data.append(data)
+    return text_data
+
```
Comment (astromz) [inline src/extract_topic.py:18]: remove extra line

File: src/extract_topic.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    def build_analyzer(self):
+        lemm = WordNetLemmatizer()
+        analyzer = super(LemmaCountVectorizer, self).build_analyzer()
+        return lambda doc: (lemm.lemmatize(w) for w in analyzer(doc))
+
+
+def process_text2vec(text):
+
+    
+    tf_vectorizer = LemmaCountVectorizer(max_df=0.95,
```
Comment (astromz) [inline src/extract_topic.py]: what's `tf` here? Use another name if it does not refer to tensforflow for better readability.

File: src/get_ocr.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+s3 = boto3.resource('s3', region_name='us-west-2')
+bucket = s3.Bucket('newspaper-ru-insight')
+object = bucket.Object('13_01_000001.tif')
+
+file_stream = io.BytesIO()
+object.download_fileobj(file_stream)
+print(file_stream)
+image = Image.open(file_stream)
+
+
```
Comment (astromz) [inline src/get_ocr.py]: Is this finished?

File: src/pipeline.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import sys
+
+import extract_entity
+import extract_topic
+
+
```
Comment (astromz) [inline src/pipeline.py:5]: remove extra line. Should only have two lines between new functions and classes.

File: src/extract_topic.py [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+        with open(file_name) as f:
+            #data = f.readlines()
+            data = " ".join(line.strip() for line in f)
+            text_data.append(data)
+    return text_data
+
+
+
+class LemmaCountVectorizer(CountVectorizer):
+    def build_analyzer(self):
```
Comment (astromz) [inline src/extract_topic.py:22]: Add class docstr to explain what it does.

File: src/extract_topic.py [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import glob
+from nltk.stem import WordNetLemmatizer
+from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer
+from sklearn.decomposition import NMF, LatentDirichletAllocation
+
+
+
+
+def generate_text_li(text_dir):
+    text_li = glob.glob(text_dir+'/*.txt')
```
Comment (astromz) [inline src/extract_topic.py]: Use a arg name that's self explanatory. `text_li` is a bit confusing.

File: src/extract_topic.py [hunk 10]
```diff
@@ -0,0 +1,65 @@
+import nltk
+import glob
+from nltk.stem import WordNetLemmatizer
+from sklearn.feature_extraction.text import TfidfVectorizer, CountVectorizer
+from sklearn.decomposition import NMF, LatentDirichletAllocation
+
```
Comment (astromz) [inline src/extract_topic.py:6]: remove two extra lines

File: src/extract_entity.py [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+            #data = f.readlines()
+        data = " ".join(line.strip() for line in f)
+            #text_data.append(data)
+
+    #nlp = spacy.load("en_core_web_sm")
+    nlp = en_core_web_sm.load()
+    doc = nlp(data)
+    entity = [(X.text, X.label_) for X in doc.ents]
+    return entity
+
```
Comment (astromz) [inline src/extract_entity.py:16]: add one more empty line.

**Review by astromz — COMMENTED** [review]
Thanks for your pull request. I had a quick look and provided some comments. Don't be discouraged by the comments, they are mostly just formatting issues and conventions.

**Comment by astromz** [comment]
I saw a few new commits. Could you mark my existing comments as resolved if they have been resolved? Thanks.

**Comment by ruchen01** [comment]
I have resolved all issues and marked them resolved.
